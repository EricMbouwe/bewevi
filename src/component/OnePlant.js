import PropTypes from 'prop-types';
import Logo from './Logo';

const OnePlant = ({
  plant, onError, emptyImage,
}) => (
  <div className="one-plant main">
    <div className="one-plant logo">
      <Logo />
    </div>
    <div className="one-plant top-part">

      <div className="one-plant main-image">
        {plant.image_url
          ? (
            <img
              src={plant.image_url}
              alt="plant"
              width="100"
              onError={onError}
              className="one-plant top-images"
            />
          )
          : emptyImage('one-plant top-images')}
      </div>
      <div className="one-plant top-description">

        <div className="one-plant description-part">
          Year:
          {'  '}
          {plant.year}
        </div>
        <div className="one-plant description-part">
          Common Name:
          {'  '}
          {plant.common_name || 'No common name'}
        </div>
        <div className="one-plant description-part">
          Scientific Name:
          {'  '}
          {plant.scientific_name || 'No scientific name'}
        </div>
        <div className="one-plant description-part">
          Common Family Name:
          {'  '}
          {plant.family_common_name || 'No common family name'}
        </div>
        <div className="one-plant description-part">
          Family Name:
          {'  '}
          {plant.family || 'No family name'}
        </div>
      </div>
    </div>
    <div className="one-plant bottom-part">
      <div className="one-plant one-property">
        <div>
          Fruit
        </div>
        {plant.images.fruit[0] ? (
          <img
            src={plant.images.fruit[0].image_url}
            alt={`fruit of ${plant.common_name}`}
            onError={onError}
            className="one-plant bottom-images"
          />
        ) : emptyImage('one-plant bottom-images')}
      </div>

      <div className="one-plant one-property">
        <div>
          Leaf
        </div>
        {plant.images.leaf[0] ? (
          <img
            className="one-plant bottom-images"
            src={plant.images.leaf[0].image_url}
            alt={`Leaf of ${plant.common_name}`}
            onError={onError}
          />
        ) : emptyImage('one-plant bottom-images')}
      </div>
      <div className="one-plant one-property">
        <div>
          Flower
        </div>
        {plant.images.flower[0] ? (
          <img
            className="one-plant bottom-images"
            src={plant.images.flower[0].image_url}
            alt={`Leaf of ${plant.common_name}`}
            onError={onError}
          />
        ) : emptyImage('one-plant bottom-images')}
      </div>
    </div>

  </div>

);

OnePlant.propTypes = {
  plant: PropTypes.shape().isRequired,
  onError: PropTypes.func.isRequired,
  emptyImage: PropTypes.shape().isRequired,
};

export default OnePlant;
