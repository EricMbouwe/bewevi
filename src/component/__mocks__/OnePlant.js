import PropTypes from 'prop-types';

const OnePlant = ({
  plant,
  handleError,
}) => (
  <>
    {JSON.stringify(plant)}
    <img src="#" alt="test" onError={handleError} />
  </>
);

OnePlant.propTypes = {
  plant: PropTypes.shape().isRequired,
  handleError: PropTypes.func.isRequired,
};

export default OnePlant;
