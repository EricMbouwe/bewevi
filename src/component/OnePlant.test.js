import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import { BrowserRouter } from 'react-router-dom';
import OnePlant from './OnePlant';

const handleError = jest.fn();
const plant = {
  image_url: 'www.ozovalihasan.com',
  images: {
    fruit: [{ imageUrl: 'www.ozovalihasan.com' }],
    leaf: [{ imageUrl: 'www.ozovalihasan.com' }],
    flower: [{ imageUrl: 'www.ozovalihasan.com' }],
  },
};

describe('<OnePlant />', () => {
  it('contains expected texts', () => {
    render(
      <BrowserRouter>
        <OnePlant
          plant={{
            image_url: null,
            images: {
              fruit: [null],
              leaf: [null],
              flower: [null],
            },
          }}
          handleError={handleError}

        />
      </BrowserRouter>,
    );
    expect(screen.getByText(/Fruit/i)).toBeInTheDocument();
    expect(screen.getByText(/Leaf/i)).toBeInTheDocument();
    expect(screen.getByText(/Flower/i)).toBeInTheDocument();
  });

  it('triggers onError when there are errors of img tags', () => {
    const rendered = ReactTestUtils.renderIntoDocument(
      <BrowserRouter>
        <OnePlant
          plant={plant}
          handleError={handleError}

        />
      </BrowserRouter>,
    );
    const imgs = ReactTestUtils.scryRenderedDOMComponentsWithTag(rendered, 'img');
    imgs.map(img => ReactTestUtils.Simulate.error(img));
    expect(handleError.mock.calls.length).toBe(4);
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <OnePlant
          plant={plant}
          handleError={handleError}

        />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
