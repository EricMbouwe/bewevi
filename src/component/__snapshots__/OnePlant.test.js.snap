// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OnePlant /> renders correctly 1`] = `
<div
  className="one-plant main"
>
  <div
    className="one-plant logo"
  >
    <a
      className="logo main"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo image"
        src="emptyImage.svg"
      />
      <div
        className="logo title"
      >
        vy
      </div>
    </a>
  </div>
  <div
    className="one-plant top-part"
  >
    <div
      className="one-plant main-image"
    >
      <img
        alt="plant"
        className="one-plant top-images"
        onError={[MockFunction]}
        src="www.ozovalihasan.com"
      />
    </div>
    <div
      className="one-plant top-description"
    >
      <div
        className="one-plant description-part"
      >
        Year:
          
      </div>
      <div
        className="one-plant description-part"
      >
        Common Name:
          
        No common name
      </div>
      <div
        className="one-plant description-part"
      >
        Scientific Name:
          
        No scientific name
      </div>
      <div
        className="one-plant description-part"
      >
        Common Family Name:
          
        No common family name
      </div>
      <div
        className="one-plant description-part"
      >
        Family Name:
          
        No family name
      </div>
    </div>
  </div>
  <div
    className="one-plant bottom-part"
  >
    <div
      className="one-plant one-property"
    >
      <div>
        Fruit
      </div>
      <img
        alt="fruit of undefined"
        className="one-plant bottom-images"
        onError={[MockFunction]}
      />
    </div>
    <div
      className="one-plant one-property"
    >
      <div>
        Leaf
      </div>
      <img
        alt="Leaf of undefined"
        className="one-plant bottom-images"
        onError={[MockFunction]}
      />
    </div>
    <div
      className="one-plant one-property"
    >
      <div>
        Flower
      </div>
      <img
        alt="Flower of undefined"
        className="one-plant bottom-images"
        onError={[MockFunction]}
      />
    </div>
  </div>
</div>
`;
